lldb/test/API/lang/swift/hashed_container_storing_nested_generic/TestHashedContainerStoringNestedGeneric.py

```diff
@@ -0,0 +1,16 @@
+import lldb
+from lldbsuite.test.lldbtest import *
+from lldbsuite.test.decorators import *
+import lldbsuite.test.lldbutil as lldbutil
+import unittest2
+
+
+class SwiftGenericClassInHashedContainerTest(TestBase):
+
+    @swiftTest
+    def test(self):
+        """Tests that the type of hashed container whose value type is a non-generic class declared inside a generic class can be read from instance metadata"""
+        self.build()
+        (target, process, thread, breakpoint) = lldbutil.run_to_source_breakpoint(self, 
+                "break here", lldb.SBFileSpec("main.swift"))
+        self.expect("v foo", substrs=["(a.A<Int>) foo", "dict = 1 key/value pair {"])
```

Comment:
Running this locally, this case already works.

```
(main.A<Int>) foo = 0x00006000002054a0 {
  dict = 1 key/value pair {
    [0] = (key = "key", value = 0x0000600000008040)
  }
}
```

Reply:
This one however, I expected to fail on current Swift -- I will take a look, perhaps I reduced the tests case too much. This was intended to be a stripped down version of https://github.com/apple/swift/issues/60565. Thanks for catching this.

Reply:
thanks for looking into it.

Reply:
@NuriAmari it's because the inner class is public in this test. You needed it private for that anonymous context to be there. You covered that in one of the other test cases though.

Reply:
> @NuriAmari it's because the inner class is public in this test. You needed it private for that anonymous context to be there. You covered that in one of the other test cases though.

I don't think its quite that simple -- though maybe that is also an issue. I've just reproduced the failure without the patches with the inner class still public. 

```
~/git/repro-pad/LLDB-crashes-due-to-difference-between-RemoteAST-and-Reflection-metadata $SWIFT_PATH/swiftc -g -sdk $(xcrun --show-sdk-path) Hashed.swift -o Hashed
~/git/repro-pad/LLDB-crashes-due-to-difference-between-RemoteAST-and-Reflection-metadata $LLDB_PATH/lldb Hashed
(lldb) target create "Hashed"
Current executable set to '/Users/nuriamari/git/repro-pad/LLDB-crashes-due-to-difference-between-RemoteAST-and-Reflection-metadata/Hashed' (arm64).
(lldb) b Hashed.swift:11
Breakpoint 1: where = Hashed`main + 84 at Hashed.swift:11:7, address = 0x0000000100005208
(lldb) r
Process 6080 launched: '/Users/nuriamari/git/repro-pad/LLDB-crashes-due-to-difference-between-RemoteAST-and-Reflection-metadata/Hashed' (arm64)
Process 6080 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = breakpoint 1.1
    frame #0: 0x0000000100005208 Hashed`main at Hashed.swift:11:7
   1    public class A<T> {
   2        public func addValue() {
   3            self.dict["key"] = B()
   4        }
   5        public class B {}
   6        private var dict: [String: B] = [:]
   7    }
   8
   9    let foo = A<Int>()
   10   foo.addValue()
-> 11   print("break here") // break here
Target 0: (Hashed) stopped.
(lldb) v foo
RemoteAST and runtime diverge $ss18_DictionaryStorageCySS6Hashed1AC1BCySi_GGD != $ss18_DictionaryStorageCySS6Hashed1AC1BCy_SiGGD
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.      Program arguments: /Users/nuriamari/git/swift-project/build/Ninja-RelWithDebInfoAssert/lldb-macosx-arm64/bin/lldb Hashed
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0  lldb                     0x0000000102a2a6d4 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) + 56
1  lldb                     0x0000000102a29d08 llvm::sys::RunSignalHandlers() + 112
2  lldb                     0x0000000102a2ad5c SignalHandler(int) + 304
3  libsystem_platform.dylib 0x000000018bc634a4 _sigtramp + 56
4  LLDB                     0x000000011a8b4d68 lldb_private::SwiftLanguageRuntimeImpl::GetMemberVariableOffsetRemoteAST(lldb_private::CompilerType, lldb_private::ValueObject*, llvm::StringRef) + 196
5  LLDB                     0x000000011a8b624c lldb_private::SwiftLanguageRuntimeImpl::GetMemberVariableOffset(lldb_private::CompilerType, lldb_private::ValueObject*, llvm::StringRef, lldb_private::Status*) + 700
6  LLDB                     0x000000011a8aefa0 lldb_private::SwiftLanguageRuntime::GetMemberVariableOffset(lldb_private::CompilerType, lldb_private::ValueObject*, llvm::StringRef, lldb_private::Status*) + 40
7  LLDB                     0x000000011a864dd8 lldb_private::formatters::swift::NativeHashedStorageHandler::NativeHashedStorageHandler(std::__1::shared_ptr<lldb_private::ValueObject>, lldb_private::CompilerType, lldb_private::CompilerType) + 728
8  LLDB                     0x000000011a8647ac lldb_private::formatters::swift::HashedCollectionConfig::CreateNativeHandler(std::__1::shared_ptr<lldb_private::ValueObject>, std::__1::shared_ptr<lldb_private::ValueObject>) const + 452
9  LLDB                     0x000000011a865e18 lldb_private::formatters::swift::HashedCollectionConfig::CreateHandler(lldb_private::ValueObject&) const + 1136
10 LLDB                     0x000000011a8663d0 lldb_private::formatters::swift::HashedSyntheticChildrenFrontEnd::Update() + 36
11 LLDB                     0x000000011a5bf058 lldb_private::ValueObjectSynthetic::UpdateValue() + 300
12 LLDB                     0x000000011a5ad688 lldb_private::ValueObject::UpdateValueIfNeeded(bool) + 952
13 LLDB                     0x000000011a5b09fc lldb_private::ValueObject::GetValueAsCString() + 36
14 LLDB                     0x000000011a5d6b98 lldb_private::ValueObjectPrinter::GetValueSummaryError(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&) + 164
15 LLDB                     0x000000011a5d651c lldb_private::ValueObjectPrinter::PrintValueAndSummaryIfNeeded(bool&, bool&) + 276
16 LLDB                     0x000000011a5d5d30 lldb_private::ValueObjectPrinter::PrintValueObject() + 192
17 LLDB                     0x000000011a5d74dc lldb_private::ValueObjectPrinter::PrintChild(std::__1::shared_ptr<lldb_private::ValueObject>, lldb_private::DumpValueObjectOptions::PointerDepth const&) + 608
18 LLDB                     0x000000011a5d77f4 lldb_private::ValueObjectPrinter::PrintChildren(bool, bool, lldb_private::DumpValueObjectOptions::PointerDepth const&) + 244
19 LLDB                     0x000000011a5d6ad4 lldb_private::ValueObjectPrinter::PrintChildrenIfNeeded(bool, bool) + 896
20 LLDB                     0x000000011a5d5d48 lldb_private::ValueObjectPrinter::PrintValueObject() + 216
21 LLDB                     0x000000011a5b4bdc lldb_private::ValueObject::Dump(lldb_private::Stream&, lldb_private::DumpValueObjectOptions const&) + 96
22 LLDB                     0x000000011abe7f44 CommandObjectFrameVariable::DoExecute(lldb_private::Args&, lldb_private::CommandReturnObject&) + 2112
23 LLDB                     0x000000011a6332ec lldb_private::CommandObjectParsed::Execute(char const*, lldb_private::CommandReturnObject&) + 428
24 LLDB                     0x000000011a62aae4 lldb_private::CommandInterpreter::HandleCommand(char const*, lldb_private::LazyBool, lldb_private::CommandReturnObject&) + 2240
25 LLDB                     0x000000011a62de40 lldb_private::CommandInterpreter::IOHandlerInputComplete(lldb_private::IOHandler&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&) + 676
26 LLDB                     0x000000011a56de34 lldb_private::IOHandlerEditline::Run() + 304
27 LLDB                     0x000000011a552204 lldb_private::Debugger::RunIOHandlers() + 140
28 LLDB                     0x000000011a62f000 lldb_private::CommandInterpreter::RunCommandInterpreter(lldb_private::CommandInterpreterRunOptions&) + 160
29 LLDB                     0x0000000119fed800 lldb::SBDebugger::RunCommandInterpreter(bool, bool) + 148
30 lldb                     0x0000000102a11014 Driver::MainLoop() + 2704
31 lldb                     0x0000000102a11b98 main + 2332
32 dyld                     0x0000000102c6108c start + 520
Segmentation fault: 11
```

One thing to note is, this failure and the one described in this issue: https://github.com/apple/swift/issues/60565 only fail on non-release builds -- the warnings and crashing code is gated by a `#ifndef NDEBUG`.

Reply:
I will try to refine the test such that is fails on release builds as well. If I am unable to do so, is there some value in still including the test and perhaps writing a comment?

Reply:
Modified test to fail before changes and pass after -- regardless of build type.

Reply:
awesome, relooking